opentelemetry-instrumentation/src/opentelemetry/instrumentation/_semconv.py

```diff
@@ -41,24 +46,147 @@
     SpanAttributes.NETWORK_PROTOCOL_VERSION,
     SpanAttributes.SERVER_ADDRESS,
     SpanAttributes.SERVER_PORT,
-    # TODO: Support opt-in for scheme in new semconv
-    # SpanAttributes.URL_SCHEME,
+    SpanAttributes.URL_SCHEME,
 ]
 
+_server_duration_attrs_old = [
+    SpanAttributes.HTTP_METHOD,
+    SpanAttributes.HTTP_HOST,
+    SpanAttributes.HTTP_SCHEME,
+    SpanAttributes.HTTP_STATUS_CODE,
+    SpanAttributes.HTTP_FLAVOR,
+    SpanAttributes.HTTP_SERVER_NAME,
+    SpanAttributes.NET_HOST_NAME,
+    SpanAttributes.NET_HOST_PORT,
+]
+
+_server_duration_attrs_new = [
+    _SPAN_ATTRIBUTES_ERROR_TYPE,
+    SpanAttributes.HTTP_REQUEST_METHOD,
+    SpanAttributes.HTTP_RESPONSE_STATUS_CODE,
+    SpanAttributes.HTTP_ROUTE,
+    SpanAttributes.NETWORK_PROTOCOL_VERSION,
+    SpanAttributes.URL_SCHEME,
+]
+
+_server_active_requests_count_attrs_old = [
+    SpanAttributes.HTTP_METHOD,
+    SpanAttributes.HTTP_HOST,
+    SpanAttributes.HTTP_SCHEME,
+    SpanAttributes.HTTP_FLAVOR,
+    SpanAttributes.HTTP_SERVER_NAME,
+    SpanAttributes.NET_HOST_NAME,
+    SpanAttributes.NET_HOST_PORT,
+]
+
+_server_active_requests_count_attrs_new = [
+    SpanAttributes.HTTP_REQUEST_METHOD,
+    SpanAttributes.URL_SCHEME,
+]
+
+_OTEL_SEMCONV_STABILITY_OPT_IN_KEY = "OTEL_SEMCONV_STABILITY_OPT_IN"
+
 
-def _filter_duration_attrs(attrs, sem_conv_opt_in_mode):
+class _OpenTelemetryStabilitySignalType:
+    HTTP = "http"
+
+
+class _OpenTelemetryStabilityMode(Enum):
+    # http - emit the new, stable HTTP and networking conventions ONLY
+    HTTP = "http"
+    # http/dup - emit both the old and the stable HTTP and networking conventions
+    HTTP_DUP = "http/dup"
+    # default - continue emitting old experimental HTTP and networking conventions
+    DEFAULT = "default"
+
+
+def _report_new(mode):
+    return mode.name != _OpenTelemetryStabilityMode.DEFAULT.name
+
+
+def _report_old(mode):
+    return mode.name != _OpenTelemetryStabilityMode.HTTP.name
+
+
+class _OpenTelemetrySemanticConventionStability:
+    _initialized = False
+    _lock = threading.Lock()
+    _OTEL_SEMCONV_STABILITY_SIGNAL_MAPPING = {}
+
+    @classmethod
+    def _initialize(cls):
+        with _OpenTelemetrySemanticConventionStability._lock:
+            if not _OpenTelemetrySemanticConventionStability._initialized:
+                # Users can pass in comma delimited string for opt-in options
+                # Only values for http stability are supported for now
+                opt_in = os.environ.get(_OTEL_SEMCONV_STABILITY_OPT_IN_KEY, "")
+                opt_in_list = []
```

Comment:
Do not implement like this, use the ordering provided by a numerical `Enum` to get the stability mode:

```python
opentelemetry_stability_mode = sorted(
    [
        getattr(OptIn, element) for element in set(
            environ.get(
                "OPENTELEMETRY_SEMCONV_STABILITY_OPT_IN",
                OptIn.default.value
            ).split(",")
        )
        if element in OptIn.__members__
    ],
    key=lambda x: x.value
)[0]

```

Reply:
We cannot have a numerical Enum since we need the value to represent the explicit string value in the env var. In your example, `HTTP_DUP.value` does not equal "http/dup" that will be read from envvar. Theoretically I guess we can implement a named tuple but I don't think the ordering provides much value.

Reply:
> We cannot have a numerical Enum since we need the value to represent the explicit string value in the env var. In your example, `HTTP_DUP.value` does not equal "http/dup" that will be read from envvar. Theoretically I guess we can implement a named tuple but I don't think the ordering provides much value.

That's just because `/` cannot be a character in an `Enum` attribute name, there is a workaround:

```python
from enum import Enum
from ipdb import set_trace
from os import environ


class OptIn(Enum):
    http_dup = 0
    http = 1
    default = 2


opt_in = sorted(
    [
        getattr(OptIn, element.replace("/", "_")) for element in set(
            environ.get(
                "OPENTELEMETRY_SEMCONV_STABILITY_OPT_IN",
                # OptIn.default.name
                "http/dup"
            ).split(",")
        )
        if element.replace("/", "_") in OptIn.__members__
    ],
    key=lambda x: x.value
)[0]

assert opt_in.name == "http_dup"
assert opt_in.value == 0

set_trace()
```
I find important that the objects that represent the stability type can represent their priority themselves.